pkg/actors/internal_actor.go

```diff
@@ -21,145 +21,57 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"strings"
 
-	"github.com/dapr/dapr/pkg/apphealth"
-	"github.com/dapr/dapr/pkg/config"
-	invokev1 "github.com/dapr/dapr/pkg/messaging/v1"
-	commonv1pb "github.com/dapr/dapr/pkg/proto/common/v1"
+	"github.com/dapr/dapr/pkg/actors/internal"
 )
 
 const InternalActorTypePrefix = "dapr.internal."
 
 // InternalActor represents the interface for invoking an "internal" actor (one which is built into daprd directly).
 type InternalActor interface {
 	SetActorRuntime(actorsRuntime Actors)
-	InvokeMethod(ctx context.Context, actorID string, methodName string, data []byte) (any, error)
+	InvokeMethod(ctx context.Context, actorID string, methodName string, data []byte, metadata map[string][]string) ([]byte, error)
 	DeactivateActor(ctx context.Context, actorID string) error
-	// InvokeReminder invokes reminder logic for an internal actor.
-	// Note that the DecodeInternalActorReminderData function should be used to decode the [data] parameter.
-	InvokeReminder(ctx context.Context, actorID string, reminderName string, data []byte, dueTime string, period string) error
-	InvokeTimer(ctx context.Context, actorID string, timerName string, params []byte) error
+	InvokeReminder(ctx context.Context, reminder InternalActorReminder, metadata map[string][]string) error
+	InvokeTimer(ctx context.Context, timer InternalActorTimer, metadata map[string][]string) error
 }
 
-type internalActorChannel struct {
-	actors map[string]InternalActor
+type InternalActorReminder struct {
+	ActorID string
+	Name    string
+	Data    []byte
+	DueTime string
+	Period  string
 }
 
-func newInternalActorChannel() *internalActorChannel {
-	return &internalActorChannel{
-		actors: make(map[string]InternalActor),
+func newInternalActorReminder(r *internal.Reminder) InternalActorReminder {
+	return InternalActorReminder{
+		ActorID: r.ActorID,
+		Name:    r.Name,
+		Data:    r.Data,
+		DueTime: r.DueTime,
+		Period:  r.Period.String(),
 	}
 }
 
-func (c *internalActorChannel) AddInternalActor(actorType string, actorImpl InternalActor) error {
-	// use internal type name prefixes to avoid conflicting with externally defined actor types
-	internalTypeName := actorType
-	if !strings.HasPrefix(actorType, InternalActorTypePrefix) {
-		internalTypeName = InternalActorTypePrefix + actorType
-	}
-	if _, exists := c.actors[internalTypeName]; exists {
-		return fmt.Errorf("internal actor named '%s' already exists", actorType)
-	}
-	c.actors[internalTypeName] = actorImpl
-	return nil
+type InternalActorTimer struct {
```

Comment:
Can we use single struct for `InternalActorReminder` and `InternalActorTimer`?

Reply:
I think so, it would probably help simplifying. I'll change that.

Reply:
Done